pkg/detectors/postgres/postgres.go

```diff
@@ -0,0 +1,239 @@
+		password, _ := pgURL.User.Password()
+		hostport := pgURL.Host
+		result := detectors.Result{
+			DetectorType: detectorspb.DetectorType_Postgres,
+			Raw:          []byte(username + password),
+			RawV2:        []byte(hostport + username + password),
+		}
+
+		if verify {
+			isVerified, verificationErr := verifyPostgres(&pgURL)
+			result.Verified = isVerified
+			result.SetVerificationError(verificationErr, password)
+		}
+
+		if !result.Verified && detectors.IsKnownFalsePositive(password, detectors.DefaultFalsePositives, true) {
+			continue
+		}
+		results = append(results, result)
+	}
+
+	return results, nil
+}
+
+func findUriMatches(dataStr string) url.URL {
+	var pgURL url.URL
+	for _, uri := range uriPattern.FindAllString(dataStr, -1) {
+		pgURL, err := url.Parse(uri)
+		if err != nil {
+			continue
+		}
+		if pgURL.User != nil {
+			return *pgURL
+		}
+	}
+	return pgURL
+}
+
+// check if postgres is running
+func postgresRunning(hostname, port string) bool {
+	connStr := fmt.Sprintf("host=%s port=%s sslmode=disable", hostname, port)
+	db, err := sql.Open("postgres", connStr)
+	if err != nil {
+		return false
+	}
+	defer db.Close()
+	return true
+}
+
+func findComponentMatches(verify bool, dataStr string) []url.URL {
+	usernameMatches := common.UsernameRegexCheck("").Matches([]byte(dataStr))
+	passwordMatches := common.PasswordRegexCheck("").Matches([]byte(dataStr))
+	hostnameMatches := hostnamePattern.FindAllStringSubmatch(dataStr, -1)
+	portMatches := portPattern.FindAllStringSubmatch(dataStr, -1)
+
+	var pgURLs []url.URL
+
+	hosts := findHosts(verify, hostnameMatches, portMatches)
+
+	for _, username := range dedupMatches(usernameMatches) {
+		for _, password := range dedupMatches(passwordMatches) {
+			for _, host := range hosts {
+				hostname, port := strings.Split(host, ":")[0], strings.Split(host, ":")[1]
+				if combinedLength := len(username) + len(password) + len(hostname); combinedLength > 255 {
+					continue
+				}
+				postgresURL := url.URL{
+					Scheme: "postgresql",
+					User:   url.UserPassword(username, password),
+					Host:   fmt.Sprintf("%s:%s", hostname, port),
+				}
+				pgURLs = append(pgURLs, postgresURL)
+			}
+		}
+	}
+	return pgURLs
+}
+
+// if verification is turned on, and we are able to verify at least one host, return only verified hosts
+// otherwise return all hosts
+func findHosts(verify bool, hostnameMatches, portMatches [][]string) []string {
+	hostnames := dedupMatches2(hostnameMatches)
+	ports := dedupMatches2(portMatches)
+	var hosts []string
+
+	if len(hostnames) < 1 {
+		hostnames = append(hostnames, defaultHost)
+	}
+
+	if len(ports) < 1 {
+		ports = append(ports, defaultPort)
+	}
+
+	for _, hostname := range hostnames {
+		for _, port := range ports {
+			hosts = append(hosts, fmt.Sprintf("%s:%s", hostname, port))
+		}
+	}
+
+	if verify {
+		var verifiedHosts []string
+		for _, host := range hosts {
+			hostname, port := strings.Split(host, ":")[0], strings.Split(host, ":")[1]
```

Comment:
**Nit:** I don't think the Go compiler is smart enough to know this should be one call to `strings.Split`

Suggestion:
```go
parts := strings.Split(host, ":")
hostname, port = parts[0], parts[1]
```